rb/lib/selenium/webdriver/remote/bridge.rb

```diff
@@ -605,11 +552,58 @@ def find_elements_by(how, what, parent = nil)
           ids.map { |id| Element.new self, element_id_from(id) }
         end
 
+        protected
+
+        def capabilities_class
+          W3CCapabilities
+        end
+
+        def default_capabilities
+          capabilities_class.firefox
+        end
+
         private
 
-        def assert_javascript_enabled
-          return if capabilities.javascript_enabled?
-          raise Error::UnsupportedOperationError, 'underlying webdriver instance does not support javascript'
+        def bridge_module
+          Module.nesting[1]
+        end
+
+        def process_capabilities(opts)
+          capabilities = opts.delete(:desired_capabilities) || default_capabilities
+          capabilities = Capabilities.send(capabilities) if capabilities.is_a?(Symbol)
```

Comment:
I'm not sure how you would create a `Capabilities` class without making use of the browser's defaults, though. I figure if the user is passing in a value, that's the value they want and we shouldn't presume to add things to it. It's not like creating it isn't trivial.

I'm not sure how things need to be different for Firefox? From an end user standpoint, `Remote#self.w3c?` toggles whether to use `Bridge` or `W3CBridge`, which should be sufficient.

Do you have an example of something that doesn't behave as desired now?